internal/configs/version1/nginx-plus.tmpl

```diff
@@ -38,6 +38,8 @@ http {
     include       /etc/nginx/mime.types;
     default_type  application/octet-stream;
 
+    proxy_cache_path /var/cache/nginx/jwks_uri levels=1 keys_zone=jwks_uri:1m max_size=10m;
```

Comment:
I don't think we want to set this at the `http` level. This will create a cache unconditionally and it's not needed when somebody's not using `jwks`.

Reply:
[proxy_cache_path](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_cache_path) can only be in the http context. This was originally in the http context, but was set in the virtual server config file. So in cases where we deployed two Virtual Servers with this policy we could get a conflict.

e.g.
```
stderr: "2023/04/04 09:25:30 [emerg] 62#62: the same path name \"/var/cache/nginx/jwks_uri\" used in /etc/nginx/conf.d/vs_default_cafe.conf:53 and in /etc/nginx/conf.d/vs_default_webapp.conf:33\n"
finished with error: exit status 1
```

Reply:
You are right about it being there unconditionally. With OIDC this directive was only set when using the `-enable-oidc` flag in the deployment. Since proxy_cache_path can only exist in the http context we would need to add a similar flag for JWKS.

Reply:
@shaun-nx Sorry poor choice of words. I meant that we should keep it inside the VirtualServer template. We can avoid conflicts by naming the cache uniquely for the VirtualServer, maybe something like adding the policy name.

It might also be a security concern to store all in a single cache for all the VSs, we might need to look into it.

Reply:
This needs to be in the server context.  At the VS level.
Was this resolved?

Reply:
@brianehlert it can be in a VS config, but it can't exist in a server context in NGINX. If the proxy_cache_path is going to be in the VS config then the cache path's will need to be uniquely named per VirtualServer as @lucacome mentioned.

Reply:
Sorry, I phrased my question that way since a VirtualServer is analogous to a server block in the nginx.conf
Which sets a context for the cache path and for the jwks_uri request path itself.